cng/sha.go

```diff
@@ -121,6 +129,20 @@ func newSHAX(id string, flag bcrypt.AlgorithmProviderFlags, key []byte) *shaXHas
 	return sha
 }
 
+func (h *shaXHash) MarshalBinary() (data []byte, err error) {
+	state := make([]byte, len(h.obj))
+	copy(state, h.obj)
```

Comment:
I like the idea. It might be an edge case but adding a custom magic prefix does not hurt and will save someone from spending a bunch of debugging hours. As this someone will at least probably be me, let's do it!